app/main/helpers/validation_tools.py

```diff
@@ -1,31 +1,42 @@
 import os.path
-import time
-import shutil
 
+try:
+    import urlparse
+except ImportError:
+    import urllib.parse as urlparse
```

Comment:
urlparse is stdlib, but it was moved in python3. This is the most straightforward way to get a 2/3 import that I know of